src/Navbar/SearchInput.jsx

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useState } from "react";
+import { useVideoList } from "../Context/VideoLibraryContext";
+import { useNavigate } from "react-router";
+
+export const SearchInput = () => {
+    const { videoLibrary } = useVideoList();
+    const [value, setValue] = useState("");
+    const [suggestions, setSuggestions] = useState([]);
+    const [openSuggestion, setOpenSuggestion] = useState(false)
+    const navigate = useNavigate();
+    useEffect(()=>{
+        const inputValue = value.trim().toLowerCase();
+        const inputLength = inputValue.length;
+        
+        const suggestionList = inputLength === 0 ? [] : videoLibrary.videoList.filter(library =>
+          library.title.toLowerCase().search(value.toLowerCase()) !== -1 
+        );
+        setSuggestions(suggestionList.slice(0,5));
+
+    },[value])
+
+    const clickDropdownItem = (text) => {
+        setValue(text);
+        setOpenSuggestion(false)
+    }
+    const getHighlightedText = (text, highlight) => {
+        // Split on highlight term and include term into parts, ignore case
```

Comment:
This looks good. Once revisited in future, the comments on this will make things easier to understand